CLI commands execute twice in native binary: spring.shell.noninteractive.enabled=false has no effect in Spring Shell 4.0.2

## Symptom

`NativeCliSmokeTest` is failing for 5 of 8 tests. The pattern is the same in every failure: the test sees output from a *successful* command execution **and then** a `*AlreadyExistsException` from a *second* execution of the same command.

Failures (from a recent `mvn -pl acceptance-tests test` run):

- `shouldRunPromptCreateCommand` — output starts with `787ce6c3` (the new prompt's ID, printed by the first successful run), then `PromptSpecAlreadyExistsException: Prompt native-smoke/native-cli-prompt already exists` from the second run. Stack trace lands in `CliApplication.runNonInteractive(CliApplication.java:100)`.
- `shouldRunPromptShowCommand` / `shouldRunPromptChangeCommand` / `shouldRunPromptReleaseCommand` — all fail in their shared `createPrompt()` helper with the same `PromptSpecAlreadyExistsException`.
- `shouldRunRepoCloneCommand` — output is `using repo .../cloned-repos/native-smoke-<runId> now` (the return value of `RepositoryCommands.cloneRepository`, only produced after a successful clone), then `JGitInternalException: Destination path "native-smoke-<runId>" already exists and is not an empty directory` from a second clone attempt. Stack trace lands in `ShellRunnerAutoConfiguration.lambda\$springShellApplicationRunner\$0`.

`shouldRunRepoCreateCommand` and `shouldRunRepoUseCommand` accidentally pass: `RepositoryCommands.createRepository` catches `RemoteRepositoryAlreadyExistsException` and returns exit code 0; `repo use` is idempotent. `shouldRunStudioCommand` passes because it uses `startCliCommand` (long-running studio process) rather than `runCliCommand`.

## Root cause

Each native-binary invocation runs **two** `ApplicationRunner` beans:

1. Spring Shell's built-in `ShellRunnerAutoConfiguration.springShellApplicationRunner` (delegates to `NonInteractiveShellRunner`).
2. The custom `CliApplication#cliShellApplicationRunner` (instantiates its own `NonInteractiveShellRunner` and runs the args through it again).

The CLI tries to suppress (1) with:

\`\`\`java
\"spring.shell.interactive.enabled=false\",
\"spring.shell.noninteractive.enabled=false\",
\`\`\`

In Spring Shell 4.0.2, `ShellRunnerAutoConfiguration` is gated on `spring.shell.interactive.enabled` only — `spring.shell.noninteractive.enabled` is not consulted there. PR #135 (`6251447`) added these properties when an older Spring Shell was in use; the dependency bump in #108 (`dbcdb7d`) silently invalidated that suppression but the regression was masked because the smoke tests rely on exit code 0, and `repo create` / `repo use` happen to tolerate a second invocation.

The execution order is consistent across the failures (built-in runner first, custom runner second for `prompt *`; custom runner first, built-in runner second for `repo clone` — the stack trace tells you which one observed the *second* run). Either way both runners fire.

## Fix

Remove the custom `cliShellApplicationRunner` and rely on Spring Shell's built-in non-interactive runner. The custom runner only added:

- Friendlier error messages on `CommandNotFoundException` / `CommandExecutionException`.
- `--help`/`-h`/`--version`/`-V`/no-args normalisation in `normalizeArgs`.
- Explicit `applicationContext.close()` (Spring Boot already closes the context on runner completion).

If we want the error-message wrapping back, it can be reintroduced via a `CommandExceptionResolver` bean — out of scope for this fix.

## Acceptance

- `mvn -pl acceptance-tests -Dtest=NativeCliSmokeTest test` (preceded by a fresh native build of `promptlm-cli`) passes all 8 cases.
- Manual sanity check: `promptlm-cli prompt create ...` in a real workspace creates the prompt exactly once.

## Notes

- Discovered while diagnosing a failing `build-full.sh` run on 2026-05-13.

## Verification

Built a fresh `promptlm-cli` native binary on this branch (commit `787ce6c3` baseline + the fix above) and re-ran `NativeCliSmokeTest`:

| Test | Before | After |
|---|---|---|
| `shouldRunRepoCreateCommand` | PASS | PASS |
| `shouldRunPromptCreateCommand` | FAIL (PromptSpecAlreadyExistsException) | **PASS** |
| `shouldRunPromptShowCommand` | FAIL (PromptSpecAlreadyExistsException) | **PASS** |
| `shouldRunPromptChangeCommand` | FAIL (PromptSpecAlreadyExistsException) | **PASS** |
| `shouldRunPromptReleaseCommand` | FAIL (PromptSpecAlreadyExistsException) | FAIL (different cause — see below) |
| `shouldRunRepoUseCommand` | PASS | PASS |
| `shouldRunRepoCloneCommand` | FAIL (JGit "Destination path already exists") | **PASS** |
| `shouldRunStudioCommand` | PASS | PASS |

The double-execution issue is resolved — the four tests whose root cause was the second redundant pass through Spring Shell now pass.

## New / pre-existing failure surfaced by the fix

`shouldRunPromptReleaseCommand` still fails, but the cause is unrelated to this issue. With double-execution gone, the `prompt release` command now actually runs end-to-end and hits the `PreReleaseExecuteGate` introduced in #140/#141/#143:

```
Caused by: dev.promptlm.release.PreReleasePromptFailure: Pre-release execution failed for prompt 787ce6c3
    at dev.promptlm.release.PreReleaseExecuteGate.runOrThrow(PreReleaseExecuteGate.java:117)
Caused by: com.openai.errors.UnauthorizedException: 401: Incorrect API key provided: ${OPENAI*****KEY}.
```

The gate calls OpenAI for real. The webapp test path gets stubbed via `AcceptanceTestStubGateway` (`@Profile("acceptance")`, added in #141), but the native CLI smoke test launches a separate native binary that has neither the stub on its classpath nor the `acceptance` profile active. Tracking separately.

--- a/apps/promptlm-cli/src/main/java/dev/promptlm/CliApplication.java
+++ b/apps/promptlm-cli/src/main/java/dev/promptlm/CliApplication.java
@@ -17,19 +17,10 @@
 package dev.promptlm;
 
 import dev.promptlm.infrastructure.config.ObjectMapperConfiguration;
-import org.springframework.boot.ApplicationRunner;
 import org.springframework.boot.WebApplicationType;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Import;
-import org.springframework.shell.core.NonInteractiveShellRunner;
-import org.springframework.shell.core.command.CommandParser;
-import org.springframework.shell.core.command.CommandExecutionException;
-import org.springframework.shell.core.command.CommandNotFoundException;
-import org.springframework.shell.core.command.CommandRegistry;
 
 @SpringBootApplication
 @Import(ObjectMapperConfiguration.class)
@@ -44,74 +35,21 @@ public class CliApplication {
                 .properties(
                         "debug=false",
                         "spring.main.banner-mode=off",
-                        "promptlm.cli.runner.enabled=true",
-                        // Suppress Spring Shell's built-in InteractiveShellRunner /
-                        // NonInteractiveShellRunner ApplicationRunners — we drive
-                        // command dispatch ourselves via cliShellApplicationRunner.
-                        // If the built-in runners are left enabled they emit a stray
-                        // "Running in interactive mode, arguments will be ignored"
-                        // WARN onto stdout, which corrupts CLI output capture in
-                        // smoke tests (NativeCliSmokeTest#shouldRunPromptReleaseCommand).
+                        // Disable Spring Shell's InteractiveShellRunner so a `promptlm-cli`
+                        // invocation with arguments dispatches once through the built-in
+                        // NonInteractiveShellRunner instead of also entering an interactive
+                        // REPL. Previously we additionally registered a custom
+                        // `cliShellApplicationRunner` that re-dispatched every command
+                        // through a second NonInteractiveShellRunner — that caused every
+                        // CLI command to execute twice in the native binary (see #144),
+                        // surfacing as `PromptSpecAlreadyExistsException` /
+                        // `JGitInternalException: Destination path ... already exists` in
+                        // NativeCliSmokeTest. The built-in runner alone is sufficient.
                         "spring.shell.interactive.enabled=false",
-                        "spring.shell.noninteractive.enabled=false",
                         "logging.level.dev.promptlm.infrastructure.config.SerializingAppContext=ERROR",
                         "spring.autoconfigure.exclude="
                                 + "org.springframework.boot.micrometer.metrics.autoconfigure.jvm.JvmMetricsAutoConfiguration"
                 )
                 .run(args);
-    }
-
-    @Bean
-    ApplicationRunner cliShellApplicationRunner(CommandParser commandParser,
-                                                CommandRegistry commandRegistry,
-                                                @Value("${promptlm.cli.runner.enabled:false}") boolean cliRunnerEnabled,
-                                                ConfigurableApplicationContext applicationContext) {
-        return applicationArguments -> {
-            if (!cliRunnerEnabled) {
-                return;
-            }
-            String[] sourceArgs = normalizeArgs(applicationArguments.getSourceArgs());
-            try {
-                runNonInteractive(sourceArgs, commandParser, commandRegistry);
-            }
-            finally {
-                applicationContext.close();
-            }
-        };
-    }
-
-    private static void runNonInteractive(String[] sourceArgs,
-                                          CommandParser commandParser,
-                                          CommandRegistry commandRegistry) throws Exception {
-        try {
-            new NonInteractiveShellRunner(commandParser, commandRegistry).run(sourceArgs);
-        }
-        catch (CommandNotFoundException ex) {
-            throw new IllegalArgumentException("Command '%s' not found.".formatted(ex.getCommandName()), ex);
-        }
-        catch (CommandExecutionException ex) {
-            Throwable rootCause = ex;
-            while (rootCause.getCause() != null && rootCause.getCause() != rootCause) {
-                rootCause = rootCause.getCause();
-            }
-            String message = rootCause != ex
-                    ? "Error executing command: " + rootCause
-                    : "Error executing command.";
-            throw new IllegalStateException(message, ex);
-        }
-    }
-
-    private static String[] normalizeArgs(String[] sourceArgs) {
-        if (sourceArgs.length == 0) {
-            return new String[]{"help"};
-        }
-        if (sourceArgs.length == 1) {
-            return switch (sourceArgs[0]) {
-                case "--help", "-h" -> new String[]{"help"};
-                case "--version", "-V" -> new String[]{"version"};
-                default -> sourceArgs;
-            };
-        }
-        return sourceArgs;
     }
 }

--- a/apps/promptlm-cli/src/main/java/dev/promptlm/cli/PromptLmStudioLauncher.java
+++ b/apps/promptlm-cli/src/main/java/dev/promptlm/cli/PromptLmStudioLauncher.java
@@ -305,7 +305,6 @@ public class PromptLmStudioLauncher {
                             "spring.application.name=promptlm-webapp",
                             "spring.main.banner-mode=off",
                             "server.port=" + port,
-                            "promptlm.cli.runner.enabled=false",
                             "spring.shell.interactive.enabled=false",
                             "spring.shell.noninteractive.enabled=false",
                             "management.metrics.binders.jvm.enabled=false",

